ephios/core/services/notifications/types.py

```diff
@@ -549,7 +549,9 @@ def send(cls, event: Event, content: str):
                         "event_id": event.id,
                         "content": content,
                         "event_title": event.title,
-                        "event_url": participant.reverse_event_detail(event),
+                        "event_url": (
+                            participant.reverse_event_detail(event) if participant.email else None
+                        ),
```

Comment:
I think the correct solution would be to have *less* data (drop the `event_url` key and `event_title`) and calculate that when actually rendering the notifications. This solves the bug *and* moves work/queries out of the request flow *and* is more correct as the data gathering happens later *and* is not as ambiguous as the condition you proposed.